benchmark/chart.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import ast
+import csv
+import re
+import sys
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+csv.field_size_limit(sys.maxsize)
+
+root = Path(__file__).parent.absolute()
+
+
+def compare_dataframes(branch1: str, branch2: str) -> None:
+    """Compare generated data frames between branches
+
+    Generates (mostly) the markdown report for the PR comment
+
+    Returns: None
+    """
+    gains = []
+    losses = []
+    dfA = pd.read_csv(Path.joinpath(root, "outputs", f"data-{branch1}.csv"))
+    dfB = pd.read_csv(Path.joinpath(root, "outputs", f"data-{branch2}.csv"))
+
+    # Make the lists equivalent in length
+    head_count = min([len(dfA), len(dfB)])
+
+    dfA = dfA.head(head_count)
+    dfB = dfB.head(head_count)
+
+    # Remove columns to enable comparisons
+    del dfA["Time"]
+    del dfB["Time"]
+    del dfA["Total"]
+    del dfB["Total"]
+
+    comparison = dfA.compare(dfB)
+
+    with open("outputs/output.csv", "w") as f:
+        writer = csv.writer(f)
+        writer.writerow(["ID", "GAIN", "LOSS", "OPINION_ID", "--"])
+        for row in comparison.iterrows():
+            non_overlap = set(ast.literal_eval(row[1][0])) ^ set(
+                ast.literal_eval(row[1][1])
+            )
```

Comment:
Woah, fancy stuff and a scary eval. A comment please?

Reply:
this ... is the only thing I couldn't really fix but a comment will be forthcoming